src/static/login.html

```diff
@@ -325,6 +391,20 @@
                 xhr.send();
             }
 
+            function login(user, password) {
+                var headers = {
+                    "Authorization": "Basic " + btoa(utf8(user + ":" + password))
+                };
+                send_login_request("GET", headers, false);
+            }
+
+            function converse(id, msg) {
+                var headers = {
+                    "Authorization": "Login-Reply " + id + " " + btoa(utf8(msg))
+                };
+                send_login_request("GET", headers, true);
```

Comment:
My first reaction was to be concerned about the use of "GET" here for what is a non-idempotent action. However the remainder of the HTTP authentication mechanisms (other than Basic) do this too.